crates/gitbutler-operating-modes/src/lib.rs

```diff
@@ -1,38 +1,113 @@
+use std::{fs, path::PathBuf};
+
 use anyhow::{bail, Context, Result};
 use gitbutler_command_context::CommandContext;
+use serde::{Deserialize, Serialize};
+
+pub const INTEGRATION_BRANCH_REF: &str = "refs/heads/gitbutler/integration";
+pub const EDIT_BRANCH_REF: &str = "refs/heads/gitbutler/edit";
 
-/// Operating Modes:
-/// Gitbutler currently has two main operating modes:
-/// - `in workspace mode`: When the app is on the gitbutler/integration branch.
-///     This is when normal operations can be performed.
-/// - `outside workspace mode`: When the user has left the gitbutler/integration
-///     branch to perform regular git commands.
+fn edit_mode_metadata_path(ctx: &CommandContext) -> PathBuf {
+    ctx.project().gb_dir().join("edit_mode_metadata.toml")
+}
 
-const INTEGRATION_BRANCH_REF: &str = "refs/heads/gitbutler/integration";
+pub fn read_edit_mode_metadata(ctx: &CommandContext) -> Result<EditModeMetadata> {
```

Comment:
I ended up writing the code for working with edit mode in a separate crate as there were dependencies required for that, which didn't make sense for `gitbutler-operating-modes` to contain. We do need at least `read_edit_mode_metadata` to exist in this crate so it can be read when calling `operating_mode`. `write_edit_mode_metadata` could be moved to the future crate and made private.